docs/source/api-main-setup.rst

```diff
@@ -0,0 +1,167 @@
+
+
+
+
+
+Install RabbitMQ
+----------------
+
+**For Linux Users**
+
+First, run the following command:
+
+``sudo apt install rabbitmq-server``
+
+Now that rabbitmq server is installed, create an admin user with the following commands:
+
+``sudo rabbitmqctl add_user admin password``
+
+``sudo rabbitmqctl set_user_tags admin administrator``
+
+``sudo rabbitmqctl set_permissions -p / admin ".*" ".*" ".*"``
+
+Make sure that the server is up and running:
+
+``sudo systemctl stop rabbitmq-server.service``
+
+``sudo systemctl start rabbitmq-server.service``
+
+``sudo systemctl enable rabbitmq-server.service``
+
+If you are having problems creating a user or getting the server running, try the following commands:
+
+``sudo rabbitmq-plugins enable rabbitmq_management``
+
+``sudo chown -R rabbitmq:rabbitmq /var/lib/rabbitmq/``
+
+``sudo rabbitmqadmin declare queue --vhost=/ name=email``
+
+``sudo rabbitmqadmin declare queue --vhost=/ name=builds``
+
+``sudo rabbitmqadmin list queues``
+
+When you run the last command, you should see something like this: ::
+
+    +--------+----------+
+    | name  | messages  |
+    +--------+----------+
+    | builds|     0     |
+    | email |     0	|
+    +--------+----------+
+
+
+Install Ngrok
+-------------
+
+**For Linux Users**
+
+To install, run this command:
+
+``sudo snap install ngrok``
+
+To connect to your local host run this command:
+
+``ngrok http “[https://localhost:8000](https://localhost:8000)"``
+
+
+
+Setup
+~~~~~
+
+
+
+
+
+
+
+
+How Do These Programs Work Together?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following diagrams illustrate how different parts of the API interact with each other.
+
+.. image:: _static/img/RabbitMQ.png
+
+Every time the user makes a request, it is sent through microWSGI. If it is a request that will take more than a few
+seconds, it is sent off to RabbitMQ which will delegate the tasks one by one to celery. If it is short enough, HTTP will
+handle the request.
+
+.. image:: _static/img/celery.png
+
+Celery is used to build and relay tasks and make the API more efficient. Lengthy requests are sent through RabbitMQ to
+celery, which will complete them on the side. The tasks sent to celery are ones that would ruin the user experience if
+they backlogged the HTTPs request cycle. Programs like celery are used to keep the request cycle short.
+
+
+.. image:: _static/img/docker.png
+
+When you want to build a study, celery sends that request to Docker, which then sends the study static files back to celery.
```

Comment:
Add short explanation of what Docker is being used for here - i.e. to bundle up individual studies into their own neatly-siloed containers for reproducibility of the experiment functionality, protection from later changes to code, etc.?

Reply:
This would be a good place to have an endnote link to your technical reference section!

Reply:
@robo360 Hey, I'm not quite sure what Docker is doing here, could you add your insight?